plutus-chain-index/src/Plutus/ChainIndex/UtxoState.hs

```diff
@@ -0,0 +1,49 @@
+{-# LANGUAGE DerivingStrategies    #-}
+{-# LANGUAGE DerivingVia           #-}
+{-# LANGUAGE MultiParamTypeClasses #-}
+{-| The UTXO state, kept in memory by the chain index.
+-}
+module Plutus.ChainIndex.UtxoState(
+    UtxoState
+    , UtxoIndex
+    ) where
+
+import           Data.FingerTree         (FingerTree, Measured (..))
+import           Data.Monoid             (Last (..))
+import           Data.Set                (Set)
+import qualified Data.Set                as Set
+import           GHC.Generics            (Generic)
+import           Ledger                  (Slot, TxOutRef)
+import           Plutus.ChainIndex.Types (BlockId)
+
+type UtxoIndex = FingerTree UtxoState UtxoState
+instance Measured UtxoState UtxoState where
+    measure = id
+
+-- | UTXO / ledger state, kept in memory. We are only interested in the UTXO set, everything else is stored
+--   on disk. This is OK because we don't need to validate transactions when they come in.
+data UtxoState =
+    UtxoState
+        { _usUnspentOutputs        :: Set TxOutRef -- ^ Outputs that have not been spent yet
+        , _usUnmatchedSpentOutputs :: Set TxOutRef -- ^ Outputs that we have seen a spending transaction for, but not a producing transaction
+        , _usSlot                  :: Last Slot -- ^ Last slot that we have seen
+        , _usBlockId               :: Last BlockId -- ^ Last block ID
```

Comment:
Maybe `_usTip :: Last (Slot, Maybe BlockId)` is more correct? Not sure.

Reply:
Yes that's better, thanks.